clusters/build-clusters/01_cluster/openshift-monitoring/main_alertmanager.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: monitoring.coreos.com/v1
+kind: Alertmanager
+metadata:
+  labels:
+    alertmanager: main
+  name: main
+  namespace: openshift-monitoring
+spec:
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - podAffinityTerm:
+          labelSelector:
+            matchExpressions:
+            - key: alertmanager
+              operator: In
+              values:
+              - main
+          namespaces:
+          - openshift-monitoring
+          topologyKey: kubernetes.io/hostname
+        weight: 100
+  baseImage: openshift/prometheus-alertmanager
+  containers:
+  - args:
+    - -provider=openshift
+    - -https-address=:9095
+    - -http-address=
+    - -email-domain=*
+    - -upstream=http://localhost:9093
+    - '-openshift-sar={"resource": "namespaces", "verb": "get"}'
+    - '-openshift-delegate-urls={"/": {"resource": "namespaces", "verb": "get"}}'
+    - -tls-cert=/etc/tls/private/tls.crt
+    - -tls-key=/etc/tls/private/tls.key
+    - -client-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token
+    - -cookie-secret-file=/etc/proxy/secrets/session_secret
+    - -openshift-service-account=alertmanager-main
+    - -openshift-ca=/etc/pki/tls/cert.pem
+    - -openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+    - -skip-auth-regex=^/metrics
+    env:
+    - name: HTTP_PROXY
+    - name: HTTPS_PROXY
+    - name: NO_PROXY
+    image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6d952943885b8c9d972db55153206cb8ef5ee8e1d3aa4f76dec0cc616b31b4cc
+    name: alertmanager-proxy
+    ports:
+    - containerPort: 9095
+      name: web
+    resources: {}
+    terminationMessagePolicy: FallbackToLogsOnError
+    volumeMounts:
+    - mountPath: /etc/tls/private
+      name: secret-alertmanager-main-tls
+    - mountPath: /etc/proxy/secrets
+      name: secret-alertmanager-main-proxy
+  externalUrl: https://alertmanager-main-openshift-monitoring.apps.build01.ci.devcluster.openshift.com/
```

Comment:
Is this needed to be hardcoded here?

Reply:
Storage is configured by configMap now.
So you wont see the CR `alertmanager` any more.
But back to your Q ... Yes the URL is used in the UI ... IIRC ... that is the link on the UI of slack ... when you click the alerts, it goes back to the external URL ... otherwise it will use the internal service domain. That is what happens at upstream deployment because the service for alertmanager is not exposed there: k8s does not have routes.